src/Internal/FulfilledPromise.php

```diff
@@ -56,9 +56,14 @@ public function catch(callable $onRejected): PromiseInterface
         return $this;
     }
 
+    /**
+     * @param callable(): (void|PromiseInterface<void>) $onFulfilledOrRejected
+     * @return PromiseInterface<T>
+     */
     public function finally(callable $onFulfilledOrRejected): PromiseInterface
     {
         return $this->then(function ($value) use ($onFulfilledOrRejected): PromiseInterface {
+        	/** @var T $value */
```

Comment:
nit: Mixed whitespace, should use spaces only

```suggestion
            /** @var T $value */
```